flloat/base.py

```diff
@@ -47,11 +68,54 @@ def __str__(self):
         """Get the string representation."""
         return str(self.s)
 
-    def find_labels(self) -> Set[Symbol]:
+    def find_labels(self) -> Set[AtomSymbol]:
         """Return the set of symbols."""
         return {self.s}
 
 
+class QuotedFormula(Hashable):
+    """This object is a constant representation of a formula.
+
+    Quoted formulas can be used as hashable objects and for atomic symbols.
+    """
+
+    _mutable = ["_hash"]
+
+    def __init__(self, f: Formula):
+        """Initialize.
+
+        :param f: formula to represent.
+        """
+        super().__init__()
+        self.__dict__["f"] = f
```

Comment:
Isn't this equivalent to
```python
self.f = f
```
?

E.g. try the following:
```python
class X:

    def __init__(self):
        self.y = 1

x = X()
print(x.y)  # 1
print(x.__dict__)  # {'y': 1}
x.__dict__["z"] = 2
print(x.z) # 2
```

Although I think `self.f = f` is more readable, you can leave it as it is :-)

Reply:
my bad, I haven't seen the `__setattr__` below, now I understand why you did it.